runtime/ms_rest_azure/lib/ms_rest_azure/polling_state.rb

```diff
@@ -35,21 +35,35 @@ def initialize(azure_response, retry_timeout)
       update_response(azure_response.response)
       @resource = azure_response.body
 
+      case @response.status
```

Comment:
I'd not rely on that the recorded test passed because I know that we are modifying the polling in the recording. It'd be great if you can remove recordings or some VM creation and Network / the customer scenario to make sure we are good.